test/specs/commands/sign_transaction.js

```diff
@@ -0,0 +1,175 @@
+import * as given from '../../steps/1_given';
+import * as when from '../../steps/2_when';
+import * as then from '../../steps/3_then';
+
+describe('sign transaction command', () => {
+	beforeEach(setUpCommandSignTransaction);
+	Given('an action "sign transaction"', given.anAction, () => {
+		Given(
+			'a passphrase "minute omit local rare sword knee banner pair rib museum shadow juice"',
+			given.aPassphrase,
+			() => {
+				Given(
+					'a transaction in stringified JSON format',
+					given.aTransactionInStringifiedJSONFormat,
+					() => {
+						Given(
+							'vorpal is in interactive mode',
+							given.vorpalIsInInteractiveMode,
+							() => {
+								When('the action is called', when.theActionIsCalled, () => {
+									Then(
+										'it should reject with validation error and message "No transaction was provided."',
+										then.itShouldRejectWithValidationErrorAndMessage,
+									);
+								});
+							},
+						);
+						Given(
+							'the passphrase can be retrieved from its source',
+							given.thePassphraseCanBeRetrievedFromItsSource,
+							() => {
+								Given(
+									'vorpal is in non interactive mode',
+									given.vorpalIsInNonInteractiveMode,
+									() => {
+										Given(
+											'the transaction is provided via stdIn',
+											given.theTransactionIsProvidedViaStdIn,
+											() => {
+												Given(
+													'a SignTransaction returns "fakesignature"',
+													given.aSignTransactionReturns,
+													() => {
+														When(
+															'the action is called with the stringified transaction object',
+															when.theActionIsCalledWithTheStringifiedTransactionObject,
+															() => {
+																Then(
+																	'it should resolve to transaction with the signature"',
+																	then.itShouldResolveToTransactionWithTheSignature,
+																);
+																Then(
+																	'it should call SignTransaction with the transaction and the passphrase',
+																	then.itShouldCallSignTransactionWithTheTransactionAndThePassphrase,
+																);
+															},
+														);
+													},
+												);
+											},
+										);
+									},
+								);
+								Given(
+									'a SignTransaction returns "fakesignature"',
+									given.aSignTransactionReturns,
+									() => {
+										When('the action is called', when.theActionIsCalled, () => {
+											Then(
+												'it should reject with validation error and message "No transaction was provided."',
+												then.itShouldRejectWithValidationErrorAndMessage,
+											);
+										});
+										When(
+											'the action is called with the stringified error object via vorpal stdIn',
+											when.theActionIsCalledWithTheStringifiedErrorObjectViaVorpalStdIn,
+											() => {
+												Then(
+													'it should reject with validation error and message "No transaction was provided."',
```

Comment:
I think the stringified error would be hard to reject as it is, coz it could be anything.
Instead, since we added the transaction verification, it would throw there

Reply:
If you're creating a transaction and piping it to this command, but you forgot to put the amount in the correct format, wouldn't you want to be told what you did wrong?

Reply:
This is Vorpal stdin, not regular stdin, so I think it has a standard shape.